src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -743,17 +418,10 @@ case class Apply(posn: Position, fn: String, args: Array[AST]) extends AST(posn,
                   |  Expected $fn(Double)""".stripMargin)
         }
 
-      case ("log", _) => TDouble
-        args.map(_.`type`) match {
-          case Array(a: TNumeric) => TDouble
-          case Array(a: TNumeric, b: TNumeric) => TDouble
-          case other =>
-            parseError(
-              s"""invalid arguments in call to $fn: ${ other.mkString(", ") }.
-                  |  Expected either:
-                  |    $fn(Double)
-                  |    $fn(Double, Double)""".stripMargin)
-        }
+      case ("log", Array(a, b)) if a.`type`.isInstanceOf[TNumeric] && b.`type`.isInstanceOf[TNumeric] => TDouble
```

Comment:
I don't have any sort of polymorphism, bounded or not, so the naive implementation is to register each of the 16 possible pairings. 😕